a.then(function (resolve) {
    resolve();
}).catch((err)=>{
    console.error(err);
})
